src/key_schedule.rs

```diff
@@ -89,10 +90,16 @@ impl KeySchedule {
         hkdf_expand(&self.current, key_type, kind.to_bytes(), hs_hash)
     }
 
-    pub fn derive_bytes(&self, kind: SecretKind, hs_hash: &[u8]) -> Vec<u8> {
-        let payload: PayloadU8 =
-            self.derive(PayloadU8Len(self.algorithm.len()), kind, hs_hash);
-        payload.into_inner()
+    pub fn derive_logged_secret(&self, kind: SecretKind, hs_hash: &[u8],
+                                key_log: &dyn KeyLog, log_label: &str, client_random: &[u8; 32])
```

Comment:
I think that eventually, instead of taking `log_label` as a parameter, this function should compute it from `kind` inside the `if_key_log.will_log()` branch. (I shouldn't have added the `label` parameter to `will_log()`; it's main purpose is differentiating the no-logging implementation from logs that do actual logging, and that doesn't need to be done label-by-label.)